fdbclient/MultiVersionTransaction.actor.cpp

```diff
@@ -2309,6 +2354,92 @@ void MultiVersionDatabase::DatabaseState::close() {
 	});
 }
 
+namespace {
+
+const char* initializationStateToString(MultiVersionDatabase::InitializationState initState) {
+	switch (initState) {
+	case MultiVersionDatabase::InitializationState::INITIALIZING:
+		return "initializing";
+	case MultiVersionDatabase::InitializationState::INITIALIZATION_FAILED:
+		return "initialization_failed";
+	case MultiVersionDatabase::InitializationState::CREATED:
+		return "created";
+	case MultiVersionDatabase::InitializationState::INCOMPATIBLE:
+		return "incompatible";
+	case MultiVersionDatabase::InitializationState::CLOSED:
+		return "closed";
+	default:
+		ASSERT(false);
+		return "invalid_state";
+	}
+}
+
+} // namespace
+
+//
+// Generates the client-side status report for the Multi-Version Database
+//
+// The parameter dbContextStatus contains the status report generated by
+// the wrapped Native Database (from external or local client), which is then
+// embedded within the status report of the Multi-Version Database
+//
+// The overall report schema is as follows:
+// { "Healthy": <overall health status, true or false>,
+//   "InitializationState": <initialization state of the Multi-Version Database>,
+//   "InitializationError": <initialization error code, present if initialization failed>,
+//   "ProtocolVersion" : <determined protocol version of the cluster, present if determined>,
+//   "ConnectionRecord" : <connection file name or connection string>,
+//   "DatabaseStatus" : <Native Database status report, present if successfully retrieved>,
+//   "ErrorRetrievingDatabaseStatus" : <error code of retrieving status of the Native Database, present if failed>,
+//   "AvailableClients" : [
+//	    { "ProtocolVersion" : <protocol version of the client>,
```

Comment:
Formatting on this line and `DatabaseContext.h:724`.
Thanks for following up on my comments!